File: Ejercicio 3/movil.cpp
#include "movil.h"

Movil::Movil(int x, int y, int x0, int y0, int W, int H, int hacia, string ord){
  this->x = x;
  this->y = y;
  this->x0 = x0;
  this->y0 = y0;
  this->W = W;
  this->H = H;
  this->hacia = hacia;
  this->ord = ord;
}

void Movil::mover(int x0, int y0, int hacia){
  switch (hacia) {
    case 0:
      ord = "SALIR";
      cout << "Ha salido del programa." << endl;
      break;
    case 1:
      ord = "ADELANTE";
      y = y0 + 1;
      break;
    case 2:
      ord = "DERECHA";
      x = x0 + 1;
      break;
    case 3:
      ord = "ATRAS";
      y = y0 - 1;
      break;
    case 4:
      ord = "IZQUIERDA";
      x = x0 - 1;
      break;
    case 5:
      ord = "TERMINAR";
      break;
    default:
      break;
  }
}

string Movil::getOrden() const{
  return this -> ord;
}

string Movil::toString() const {
  stringstream ss;
  ss << "orden " << this -> ord;
  return ss.str();
}

int Movil::getX() const{
  return this -> x;
}

int Movil::getY() const{
  return this -> y;
}

File: Ejercicio 3/movil.py
from random import randint

class Movil:
    def __init__(self, nombre):
        self.nombre = nombre

    def __str__(self):
        return 'Movil: {}'.format(self.nombre)

    def getX(self, x0, hacia):
        if (hacia == 2):
            return x0 + 1
        if (hacia == 4):
            return x0 - 1

    def getY(self, y0, hacia):
        if (hacia == 1):
            return y0 + 1
        if (hacia == 3):
            return y0 - 1
